paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -485,6 +488,17 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                         .getString(R.string.stripe_something_went_wrong)
                 )
             }
+        } else if (
+            FeatureFlags.cvcRecollection.isEnabled &&
+            isCvcRecollectionRequired() &&
+            paymentSelection is PaymentSelection.Saved &&
+            paymentSelection.paymentMethod.type == PaymentMethod.Type.Card
```

Comment:
I _think_ we are checking these conditions in several places -- is there a way to share this? e.g. it'd be nice if, when we remove the feature flag, we could just remove it from one place

Reply:
Several are similar but slightly different. I've update `isCvcRecollectionRequired` to check the feature flag as well which removes some repetition. I can make a util function to check the payment type is Saved and of Type.Card but then the `paymentSelection` still has to be cast `PaymentSelection.Saved` which is just as redundant